interface/src/app/analyze/analysis/sampleBin.js

```diff
@@ -277,15 +280,65 @@ function($log, $cacheFactory, $q, Sample, Activity) {
       //  note: progress can be reported by returning a $promise to the caller
       // respObj.queryStatus = 'Retrieving sample activity...';
       this.rebuildHeatmapActivity(this.heatmapData.samples);
+    },
+
+    // volcano plot methods
+    getVolcanoPlotData: function() {
```

Comment:
As you may remember, I have a divided mind about re-assigning `this` to self. I don't think the reason to do so would be *readability*, exactly. The main issue, as I see it, is *correctness*. My feeling is that as long as `this` is used correctly, there is no trouble with using it and I think there are benefits to using it in an undisguised fashion.

There are six occurrences of the keyword in the `getVolcanoPlotData` method here. We use `this` three times to invoke a sibling method on the `SampleBin` object, which seems straightforward and clearly correct to me. In each of the remaining three cases, `this` is handed off as a parameter of `Array.prototype.map`, where it must be used because the callback function in the `.map` call needs it in order to properly access the `activityCache` property of the `SampleBin` instance. I suppose that is the one case where it *might* seem confusing, but now that I've learned how JavaScript's `this` actually works, I am no longer confused by it.

Clear as mud, I guess. What do you think, would the code be easier to follow if `this` were re-assigned to, say, `SampleBin`, which is what it is? If we do that, then we're masking the outer `SampleBin` and maybe *that's* even more confusing. See what I mean?